frontend/dev-mode/src/components/ChatPrompt.vue

```diff
@@ -0,0 +1,237 @@
+<script setup lang="ts">
+import { vIntersectionObserver } from '@vueuse/components'
+import { useResizeObserver, type RemovableRef } from '@vueuse/core'
+import { ref, onMounted, inject, type Ref } from 'vue'
+
+import SettingsModal from './SettingsModal.vue'
+
+import LoadingState from '@/components/LoadingState.vue'
+import StatsValues, { type Stats } from '@/components/StatsValues.vue'
+import CopyTextButton from '@/components/ui/CopyTextButton.vue'
+import MarkdownContent from '@/components/ui/MarkdownContent.vue'
+import Textarea from '@/components/ui/Textarea.vue'
+import {
+  type Session,
+  type Parameters,
+  DEFAULT_SESSION,
+  DEFAULT_PARAMS_VALUES,
+  type TranscriptMessage
+} from '@/composables/useLlama'
+
+const message = ref('')
+const shouldAutoScroll = ref(true)
+const resultsContainer = ref(null)
+const messageInput = ref<{ inputRef: HTMLInputElement } | null>(null)
+const isSettingsModalOpen = ref(false)
+
+const parameters = inject<RemovableRef<Parameters>>('parameters', ref(DEFAULT_PARAMS_VALUES))
+
+const session = inject<Ref<Session>>('session', ref(DEFAULT_SESSION))
+const stats = inject('stats', {} as Stats)
+const isGenerating = inject('isGenerating', false)
+const isPending = inject('isPending', false)
+const isChatStarted = inject('isChatStarted', false)
+const runChat = inject('runChat', (message: string) => message)
```

Comment:
these fallback functions are only here to make TS happy, this will never happen because these injects are passed by the parent high order component, which means that the app will break if the provided functions / values are not available, but ill add these exits just get this PR pushed :)

Reply:
also, the typing cant be `runChat` so it'll be slightly different than your suggestion, it'll be a generic function